<a href="https://colab.research.google.com/github/mikkeljais/sds_tddl_2020/blob/master/exercise4part1EX.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
%matplotlib inline
import pandas as pd
import numpy as np
import matplotlib.pylab as plt
from scipy.interpolate import interp1d
import keras
from keras.models import Sequential
from keras.layers import Dense, Activation

Using TensorFlow backend.


In [2]:
#load train and test data
train=pd.read_csv('/content/drive/My Drive/exam_data/train.csv')
test=pd.read_csv('/content/drive/My Drive/exam_data/test.csv')

train.head()

Unnamed: 0  ... label
0       19769  ...     0
1      127049  ...     1
2      211740  ...     1
3      116665  ...     1
4       19867  ...     1

[5 rows x 11 columns]

In [0]:
#defining train and test variables
x_train=train['text']
y_train=train['label']
x_test=test['text']
y_test=test['label']

val_dat = (np.array([i for i in x_test.values]), y_test.values)

In [4]:
import tensorflow as tf
import tensorflow_hub as hub
#using the USE classifier
module_url = "https://tfhub.dev/google/universal-sentence-encoder/4"
#the model is defined and training is performed on the train dataset

model = tf.keras.models.Sequential()

model.add(hub.KerasLayer(module_url,input_shape=[],dtype=tf.string,trainable=False)) 

#define extra layers
model.add(tf.keras.layers.Dense(256, activation='relu'))
model.add(tf.keras.layers.Dense(1, activation='sigmoid'))

#compile model and fitting the model
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'] )
model.fit(x_train.values, 
            y_train.values, 
            epochs=10)

Epoch 1/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.1917 - accuracy: 0.9229
Epoch 2/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.1670 - accuracy: 0.9330
Epoch 3/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.1565 - accuracy: 0.9375
Epoch 4/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.1448 - accuracy: 0.9441
Epoch 5/10
1563/1563 [==============================] - 26s 17ms/step - loss: 0.1339 - accuracy: 0.9490
Epoch 6/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.1216 - accuracy: 0.9540
Epoch 7/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.1104 - accuracy: 0.9600
Epoch 8/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.0993 - accuracy: 0.9651
Epoch 9/10
1563/1563 [==============================] - 27s 17ms/step - loss: 0.0882 - accuracy: 0.9701
Epoch 10/10
1563/1563 [==============================] - 27s 17m

So we can see that the model achieves around 97% accuracy in correcctly labeling whether the given Trustpilot review amounts in a good or a bad rating.